### ISSUE TITLE
When font size is turned off in theme.json, the typography panel does not completely hide the font size option.

### ISSUE THREAD

**Issue by carolinan** [WordPress/gutenberg#59732]
### Description

Developers can use theme.json to enable font sizes site wide. They can also remove them for a single block by setting `"customFontSize": false` and leaving the fontSizes array empty:  `"fontSizes": []`. 

This removes the font size control from the **global styles** block typography panel: Site Editor > Styles Sidebar > Blocks > Block name > Typography;
But not from the Typography panel when an individual block is selected.

1.  The font size menu item is still visible, and selected, in the "Typography options" menu.
2. The font size is the default visible control in the panel, and when it is turned off, the Typography panel is expanded but empty.
3. A font size that is already set on a block, for example in a template or pattern, can be reset from the Typography options  menu. I am not sure if this is intentional or a bug. - After the discussion about resetting the shadows, I was under the impression that only the global styles should be resettable if a feature is not supported. 

### Step-by-step reproduction instructions

Add the following to the `settings` section of theme.json:
```
		"blocks": {
			"core/heading": {
				"typography": {
					"fontSizes": [],
					"customFontSize": false
				}
			}
		}
```

Open the block editor or site editor and add a heading block.
Open the Typography panel in the block settings panel (the inspector). Check if the font size options are present.

### Screenshots, screen recording, code snippet

In this image, the Typography panel is expanded by default, but empty:
<img width="575" alt="The Typography panel with a partially removed font size option" src="https://github.com/WordPress/gutenberg/assets/7422055/7fbeacca-b8ac-43b9-9e5d-44f7c89821fb">

### Environment info

WordPress 6.4.3, 6.5 RC.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by kraftner** [comment]
I just stumbled onto this. Just some additional observations:

There seem to be issues that are at least closely related: #50330 and #45273

Also this is a regression in WP 6.5 since I was able to completely disable all typography features before by having these settings in `settings.typrography` in `theme.json`:

```
"typography": {
	"fontSizes": [],
	"customFontSize": false,
	"defaultFontSizes": false,
	"writingMode": false,
	"dropCap": false,
	"textDecoration": false,
	"letterSpacing": false,
	"textTransform": false,
	"fontWeight": false,
	"fontStyle": false,
	"lineHeight": false
},
```

Another curious thing I noticed is that in 6.4 it works in the block editor when editing post content: The Typography panel is gone.
But when editing the globals styles it never goes away.

In 6.5 it stays in both cases.

---

I then looked through the codebase but wasn't really sure where to look. One issue that stood out was that at some point the panels for global styles and block styles got merged. Considering the difference in behavior described above I am guessing this could be related: #47356

**Comment by kraftner** [comment]
I investigated this further and what I noticed are basically two separate issues.

The issue above, which should already be fixed with https://github.com/WordPress/gutenberg/pull/58409/files#diff-def0802a58e1512057e07b22af00a628023b325ea4bdf28a87b322529405cdf2R54-R59

And what [I described above](https://github.com/WordPress/gutenberg/issues/59732#issuecomment-2079622922) is another issue tracked in #61231

**Comment by annezazu** [comment]
Wanted to bring attention to the work done to Add defaultFontSizes theme.json and how that allows theme authors to disable showing the default font sizes and show only custom options: https://github.com/WordPress/gutenberg/pull/58409 Does this resolve the concern or do you want an ability to completely remove any font size typography controls? Can't tell if I'm missing something here :)

**Comment by carolinan** [comment]
I have retested, and this is indeed fixed in 6.6.
